fix(cli): support sketches with custom main file names

avinxshKD opened the pull request:
## Problem
CLI rejected sketches with custom main file names (e.g., `sketch/main.pde`) even though the IDE supports them via `sketch.properties`.

## Solution
Updated CLI to check `sketch.properties` for a `main` property before falling back to default naming, matching IDE behavior from `Sketch.findMain()`.

## Changes
- `Commander.java`: Added Settings import and custom main file detection logic
- `CommanderTest.java`: Added unit tests for default and custom main file handling

Fixes #1219

avinxshKD committed:
fix(cli): support sketches with custom main file names

Previously, the CLI only accepted sketches where the main .pde file
matched the sketch folder name (e.g., sketch/sketch.pde). This caused
issues when users renamed their main file in the IDE, which stores
the custom filename in sketch.properties.

Now the CLI checks sketch.properties for a 'main' property before
falling back to the default naming convention, matching the IDE's
behavior implemented in Sketch.findMain().

Fixes #1219

SableRaf commented on java/src/processing/mode/java/Commander.java:
```diff
@@ -145,7 +146,24 @@ public Commander(String[] args) {
+        }
+        
         if (!pdeFile.exists()) {
           complainAndQuit("Not a valid sketch folder. " + pdeFile + " does not exist.", true);
```
@Stefterv Maybe instead of rejecting the folder when folderName.pde is missing, the CLI could handle the following two two cases:

1.	Opening a folder with multiple .pde files but no explicit main file: ask the user whether they want to rename the folder OR choose which .pde should act as the main one, then generate sketch.properties accordingly.
	
2.	Opening a single .pde file inside a folder without an explicit main file: treat that file as the main one and create sketch.properties to match.

That would be closer to the PDE’s behavior. What do you think?

avinxshKD committed:
test: add CLI test for custom main file support

Added testSketchWithCustomMainFile() to CLITest.kt as requested by maintainer. This test provides a placeholder for manual testing of sketches with custom main files specified in sketch.properties.

Follows the same pattern as existing CLI tests (testLSP, testLegacyCLI) and is intended to be run manually in IntelliJ IDEA.

avinxshKD committed:
test: convert to automated CLI test with temp directory

Converted testSketchWithCustomMainFile() from manual to automated test. Now creates a temporary sketch folder with custom main file and sketch.properties, then tests the CLI build command.

Follows the pattern from SchemaTest.kt using Files.createTempDirectory() and automatic cleanup.

avinxshKD commented:
@Stefterv Hey, pls review and is there anything you think I should add or improve? If yes, I’ll work on it after my exams. Thanks
